src/source/Crypto/Dtls.h

```diff
@@ -208,9 +208,17 @@ INT32 dtlsSessionSendCallback(PVOID, const unsigned char*, ULONG);
 INT32 dtlsSessionReceiveCallback(PVOID, unsigned char*, ULONG);
 VOID dtlsSessionSetTimerCallback(PVOID, UINT32, UINT32);
 INT32 dtlsSessionGetTimerCallback(PVOID);
+
+#if MBEDTLS_VERSION_NUMBER >= 0x03000000
+VOID dtlsSessionKeyDerivationCallback(PVOID customData, mbedtls_ssl_key_export_type secret_type, const unsigned char* pMasterSecret,
+                                      SIZE_T pMasterSecretLen, const unsigned char clientRandom[MAX_DTLS_RANDOM_BYTES_LEN],
+                                      const unsigned char serverRandom[MAX_DTLS_RANDOM_BYTES_LEN], mbedtls_tls_prf_types tlsProfile);
+#else
 INT32 dtlsSessionKeyDerivationCallback(PVOID, const unsigned char*, const unsigned char*, ULONG, ULONG, ULONG,
                                        const unsigned char[MAX_DTLS_RANDOM_BYTES_LEN], const unsigned char[MAX_DTLS_RANDOM_BYTES_LEN],
                                        mbedtls_tls_prf_types);
+#endif
```

Comment:
Suggest to add an error if MBEDTLS_VERSION_NUMBER is not defined?

```
#ifndef MBEDTLS_VERSION_NUMBER
#error "MBEDTLS_VERSION_NUMBER is not defined"
#endif
```

Reply:
Maybe another suggestion would be to create a new macro so that we don't have `0x03000000` everywhere?

```
#define MBEDTLS_BEFORE_V3 (MBEDTLS_VERSION_NUMBER < 0x03000000)
```

Reply:
Makes sense. Will add a common macro in Dtls.h under `KVS_USE_MBEDTLS`.